File: src/paths.ts
import { homedir } from "node:os";
import { join } from "node:path";

export const DEFAULT_APP = "/Applications/ChatGPT.app";
export const USER_ROOT = join(homedir(), ".incodex");
export const BACKUP_DIR = join(USER_ROOT, "backup");
export const STATE_PATH = join(USER_ROOT, "install-state.json");
export const LIVE_PREV = join(USER_ROOT, "ChatGPT.app.pre-live");
export const LIVE_RECORD_PATH = join(USER_ROOT, "live-install.json");
export const INSTALLATIONS_DIR = join(USER_ROOT, "installations");
export const INCOGNITO_HOME = join(USER_ROOT, "incognito-home");
export const INCOGNITO_CHROMIUM = join(USER_ROOT, "incognito-chromium");
export const MAIN_NAME = "incodex-main.cjs";
export const PRELOAD_NAME = "incodex-preload.cjs";
export const SAFE_HOME_NAME = "incodex-safe-home.cjs";
export const IPC_GUARD_NAME = "incodex-ipc-guard.cjs";
export const INSTANCE_NAME = "incodex-instance.cjs";

export const ASAR_REL = "Contents/Resources/app.asar";
export const PLIST_REL = "Contents/Info.plist";
export const LOADER_NAME = "incodex-loader.cjs";
export const INJECT_NAME = "incodex-inject.js";
export const MARKER_KEY = "__incodex";

File: crates/incodex-core/src/paths.rs
use std::path::{Path, PathBuf};

pub const DEFAULT_APP: &str = "/Applications/ChatGPT.app";
pub const RUNTIME_DIR_NAME: &str = "runtime";
pub const RUNTIME_CURRENT_NAME: &str = "current.json";
pub const ASAR_REL: &str = "Contents/Resources/app.asar";

pub fn default_app() -> PathBuf {
    PathBuf::from(DEFAULT_APP)
}

pub fn home_dir() -> PathBuf {
    std::env::var_os("HOME")
        .filter(|value| !value.is_empty())
        .map(PathBuf::from)
        .unwrap_or_else(|| PathBuf::from("/"))
}

pub fn user_root() -> PathBuf {
    user_root_from(&home_dir())
}

pub fn user_root_from(home: &Path) -> PathBuf {
    home.join(".incodex")
}
